newfile.go

```diff
@@ -0,0 +1,175 @@
+	for index, item := range books {
+		if item.ID == params["id"] {
+			books = append(books[:index], books[index+1:]...)
+			var book Book
+			_ = json.NewDecoder(r.Body).Decode(&book)
+			books = append(books, book)
+			json.NewEncoder(w).Encode(book)
+			fmt.Fprintf(w, "Data of the current entry has been updated")
+			return
+		}
+	}
+	json.NewEncoder(w).Encode(books)
+}
+func deleteBook(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range books {
+		if item.ID == params["id"] {
+			books = append(books[:index], books[index+1:]...)
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(books)
+	fmt.Fprintf(w, "The current entry hass been deleted")
+}
+
+func getUsers(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	json.NewEncoder(w).Encode(users)
+}
+
+func getUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for _, item := range users {
+		if item.Srno == params["srno"] {
+			json.NewEncoder(w).Encode(item)
+			return
+		}
+	}
+	// json.NewEncoder(w).Encode(&Book{})
+	fmt.Fprintf(w, "No Data available")
+}
+
+func createUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	var user User
+	_ = json.NewDecoder(r.Body).Decode(&user)
+	user.Srno = strconv.Itoa(rand.Intn((10000000)))
+	users = append(users, user)
+	json.NewEncoder(w).Encode(user)
+	fmt.Fprintf(w, "New Entry created and added to the database")
+}
+
+func updateUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range users {
+		if item.Srno == params["srno"] {
+			users = append(users[:index], users[index+1:]...)
+			var user User
+			_ = json.NewDecoder(r.Body).Decode(&user)
+			users = append(users, user)
+			json.NewEncoder(w).Encode(user)
+			fmt.Fprintf(w, "Data of the current entry has been updated")
+			return
+		}
+	}
+	json.NewEncoder(w).Encode(users)
+}
+
+func deleteUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+	params := mux.Vars(r)
+	for index, item := range users {
+		if item.Srno == params["srno"] {
+			users = append(users[:index], users[index+1:]...)
+			break
+		}
+	}
+	json.NewEncoder(w).Encode(users)
+	fmt.Fprintf(w, "The current entry hass been deleted")
+}
+
+func main() {
+	r := mux.NewRouter()
+
+	books[0] = Book{ID: "1", ISBN: "7741852", Title: "Char",
+		Author: &Author{Firstname: "Sachin", Lastname: "Shinde"}}
+	books[1] = Book{ID: "2", ISBN: "9963852", Title: "Pahunchaya",
+		Author: &Author{Firstname: "Virat", Lastname: "Patil"}}
+	users[0] = User{Srno: "1", FirstName: "Chinmay",
+		LastName: "Ponting", Bookspurchased: 0, Amount: 0}
+	users[1] = User{Srno: "2", FirstName: "Rushi", LastName: "Gilchrist",
+		Bookspurchased: 0, Amount: 0}
```

Comment:
We can directly initialise books variable by values since it is static values

Reply:
Okk